volume/local/local.go

```diff
@@ -388,7 +389,7 @@ func (v *localVolume) saveOpts() error {
 	if err != nil {
 		return err
 	}
-	err = os.WriteFile(filepath.Join(v.rootPath, "opts.json"), b, 0o600)
+	err = ioutils.AtomicWriteFile(filepath.Join(v.rootPath, "opts.json"), b, 0o600)
```

Comment:
I had to double-check if using this here could cause issues if `/var/lib/docker/volumes` was on a different filesystem (I know that's a scenario I've seen, where that path was mounted from a different device), but it looks like it uses the current directory as path for the temporary files it creates, so I think we're good; https://github.com/moby/moby/blob/11179de64c6e4a1f770caae2c975911077044be7/pkg/ioutils/fswriters.go#L14-L17